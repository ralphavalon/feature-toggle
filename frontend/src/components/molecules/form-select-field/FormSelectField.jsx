import React, { useState, useEffect } from 'react';
import { Form } from 'react-bootstrap';
import PropTypes from 'prop-types';

import SelectMulti from '../../atoms/select-multi';

const FormSelectField = ({
  label, name, className,
  disabled, value, placeholder,
  options, invalidMessage
}) => {

  const [fieldValue, setFieldValue] = useState(value);
  useEffect(() => setFieldValue(value), [value]);

  return (
    <Form.Group className={className}>
      <Form.Label>{label}</Form.Label>
      <SelectMulti name={name} disabled={disabled} value={fieldValue} onChange={e => setFieldValue(e)} options={options}
        placeholder={placeholder}/>
      {invalidMessage && <Form.Control.Feedback type="invalid" className="d-block">{invalidMessage}</Form.Control.Feedback>}
    </Form.Group>
  );
};

FormSelectField.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  inputProps: PropTypes.object,
  options: PropTypes.array,
  value: PropTypes.array,
};

export default FormSelectField;