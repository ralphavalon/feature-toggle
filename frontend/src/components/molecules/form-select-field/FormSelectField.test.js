import React from 'react';
import { shallow } from 'enzyme';

import FormSelectField from './FormSelectField';

afterEach(() => {
  jest.clearAllMocks();
});

const formSelectFieldRequiredProps = {
  label: 'Label',
  name: 'field-name'
};

let formSelectFieldOptionalProps = {
  className: 'className',
  disabled: true,
  inputProps: { rows: 3 },
  inputType: 'textarea',
  value: ['value'],
  options: ['option']
};

describe('FormSelectField', () => {
  it('renders correctly with required props', () => {
    const wrapper = shallow(<FormSelectField {...formSelectFieldRequiredProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('renders correctly with required props and optional props', () => {
    const wrapper = shallow(<FormSelectField {...{...formSelectFieldRequiredProps, ...formSelectFieldOptionalProps}} />);
    expect(wrapper).toMatchSnapshot();
  });
})

