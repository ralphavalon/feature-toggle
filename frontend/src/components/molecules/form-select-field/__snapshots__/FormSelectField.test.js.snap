// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelectField renders correctly with required props 1`] = `
<FormGroup>
  <FormLabel
    column={false}
    srOnly={false}
  >
    Label
  </FormLabel>
  <SelectMulti
    name="field-name"
    onChange={[Function]}
  />
</FormGroup>
`;

exports[`FormSelectField renders correctly with required props and optional props 1`] = `
<FormGroup
  className="className"
>
  <FormLabel
    column={false}
    srOnly={false}
  >
    Label
  </FormLabel>
  <SelectMulti
    disabled={true}
    name="field-name"
    onChange={[Function]}
    options={
      Array [
        "option",
      ]
    }
    value={
      Array [
        "value",
      ]
    }
  />
  <Feedback
    className="d-block"
    type="invalid"
  >
    invalidMessage
  </Feedback>
</FormGroup>
`;
